service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/AddGatekeepingOrderController.java

```diff
@@ -9,162 +9,224 @@
     }
 
-    //Need to redesign for upload + urgent upload routes
-    @PostMapping("/about-to-submit")
-    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
-        @RequestBody CallbackRequest callbackRequest) {
-        CaseDetails caseDetails = callbackRequest.getCaseDetails();
-        CaseData caseData = service.updateStandardDirections(caseDetails);
-        GatekeepingOrderEventData eventData = caseData.getGatekeepingOrderEventData();
-
-        StandardDirectionOrder gatekeepingOrder = service.buildBaseGatekeepingOrder(caseData);
-
-        GatekeepingOrderSealDecision gatekeepingOrderSealDecision = eventData.getGatekeepingOrderSealDecision();
-
-        if (gatekeepingOrderSealDecision.getOrderStatus() == SEALED) {
-            //generate document
-            Document document = service.buildDocument(caseData);
+    @PostMapping("/prepare-decision/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleGenerateDraftMidEvent(@RequestBody CallbackRequest request) {
+        final CaseData caseData = orderService.updateStandardDirections(request.getCaseDetails());
+        final CaseDetailsMap data = caseDetailsMap(request.getCaseDetails());
 
-            gatekeepingOrder = gatekeepingOrder.toBuilder()
-                .dateOfIssue(formatLocalDateToString(gatekeepingOrderSealDecision.getDateOfIssue(), DATE))
-                .unsealedDocumentCopy(gatekeepingOrderSealDecision.getDraftDocument())
-                .orderDoc(buildFromDocument(document))
-                .build();
+        if (caseData.getGatekeepingOrderRouter() == URGENT) {
+            throw new UnsupportedOperationException("The prepare-decision callback does not support urgent route");
+        }
 
-            List<DocmosisTemplates> docmosisTemplateTypes = service.getNoticeOfProceedingsTemplates(caseData);
+        data.put("gatekeepingOrderSealDecision", orderService.buildSealDecision(caseData));
 
-            List<Element<DocumentBundle>> nop = nopService.uploadAndPrepareNoticeOfProceedingBundle(
-                caseData, docmosisTemplateTypes
-            );
+        if (caseData.getGatekeepingOrderRouter() == SERVICE) {
+            data.put("standardDirections", caseData.getGatekeepingOrderEventData().getStandardDirections());
+        }
+        return respond(data);
+    }
 
-            caseDetails.getData().put("noticeOfProceedingsBundle", nop);
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest request) {
+
+        final CaseData caseData = orderService.updateStandardDirections(request.getCaseDetails());
+        final CaseDetailsMap data = caseDetailsMap(request.getCaseDetails());
+        final GatekeepingOrderSealDecision decision = caseData.getGatekeepingOrderEventData()
+            .getGatekeepingOrderSealDecision();
+
+        final GatekeepingOrderRoute sdoRouter = caseData.getGatekeepingOrderRouter();
+
+        switch (sdoRouter) {
```

Comment:
i would rather say that this is about responsibility of layers, i think service should operate on domain object (accept domain object and return domain object) and should not bother how things are serialised (controller is responsible for serialisation/deserialisation), putting things into map is controlling how things are serialised. But this is theory of course. In practice it could be easier when service controls is, but here it is just one filed so i think it can stay